sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -335,6 +358,19 @@ trait Row extends Serializable {
   def getAs[T](fieldName: String): T = getAs[T](fieldIndex(fieldName))
 
   /**
+   * Returns the value of a given fieldName as a scala.util.Try object.
+   */
+  def attempt[T](fieldName: String): Try[T] = Try(getAs[T](fieldIndex(fieldName)))
```

Comment:
As I said on the other PR, I don't understand the case where you want to hide exceptions while pulling values out of a `Row`.  It seems the only time you should get an exception is when you are trying to extract a column value using the wrong type.  This seems like a structural problem with your code (its not dirty data or something transient, its going to fail for every single row).  As such, I don't think that this is a good addition to the API.